### PR TITLE
ci: merge staging to master

### PR DESCRIPTION
**Pull request by maxwell-aiden**
This is an automatic PR generated by the pipeline CI/CD. This will be automatically fast-forward merged if successful.

### CODE REVIEW

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1214433690 for 4df21fb5ca5a36fd828156070afca704820f03d8

https://gitlab.com/MatrixAI/open-source/js-quic/-/pipelines/1214433690

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 1214433690 for 4df21fb5ca5a36fd828156070afca704820f03d8

https://gitlab.com/MatrixAI/open-source/js-quic/-/pipelines/1214433690

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -220,12 +220,16 @@ build:macos:
     - hash -r
     - npm install --ignore-scripts
     - export PATH="$(npm root)/.bin:$PATH"
+    - export PATH="$(npm root)/.bin:$PATH"
```
Comment (CMCDragonkai) [inline .gitlab-ci.yml:223]: Is this duplicated @tegefaulkes?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -220,12 +220,16 @@ build:macos:
...
     # Gitlab MacOS already has rust installed
     # This will override the system installed rust
     - export PATH="$HOME/.cargo/bin:$PATH"
     # Build x64 and arm64 separately
     - npm run prebuild --verbose -- --arch x64 --production
     - npm run prebuild --verbose -- --arch arm64 --production
+    - lipo -create -output prebuild/quic-darwin-x64+arm64.node prebuild/quic-darwin-arm64.node prebuild/quic-darwin-x64.node
+    # Force remove any downloaded native bindings, so we only test with our builds
+    - rm -rf node_modules/@matrixai/quic-*
```
Comment (CMCDragonkai) [inline .gitlab-ci.yml:232]: What is `lipo` and does it come from `nix-shell` or something?

File: .gitlab-ci.yml (hunk 1 above)
Comment (tegefaulkes) [reply .gitlab-ci.yml:223]: Ah, that snuck in, I'll fix it.

File: .gitlab-ci.yml (hunk 2 above)
Comment (tegefaulkes) [reply .gitlab-ci.yml:232]: Lipo comes with the mac runner and I think it's on any mac. its the standard tool for creating universal mac binaries buy combining any number of architecture specific binaries.

File: .gitlab-ci.yml (hunk 1 above)
Comment (tegefaulkes) [reply .gitlab-ci.yml:223]: Fixed. in `1.2.5`.

**Pull request merged**